src/main/java/org/mskcc/cbio/oncokb/util/ObjectUtil.java

```diff
@@ -0,0 +1,19 @@
+package org.mskcc.cbio.oncokb.util;
+
+import org.mskcc.cbio.oncokb.service.dto.useradditionalinfo.AdditionalInfoDTO;
+
+public class ObjectUtil {
+    
+    /**
+     * Checks if a AdditionalInfoDTO is empty or null. AdditionalInfoDTO is empty if its fields are all empty as well.
+     * 
+     * @param additionalInfoDTO the user's additionalInfoDTO
+     * @return true if null or empty, otherwise false
+     */
+    public static boolean isUserAdditionalInfoEmpty(AdditionalInfoDTO additionalInfoDTO) {
+        if (additionalInfoDTO == null) {
+            return true;
+        }
+        return additionalInfoDTO.getTrialAccount() == null && additionalInfoDTO.getUserCompany() == null;
```

Comment:
This still does not give any warning nor error if properties of AdditionalInfo changes.
Can we do Gson toJson and check the String is "{}"? Or similar idea, sounds a bit hacky though.

Reply:
Then we can even take any Object and convert it to Json? name could be `isObjectEmpty`?

Reply:
Gson might throw exception though, might need to handle that. I'm not sure.